### DISCUSSION TITLE
Is there a way to add new task via SQL directly?

### DISCUSSION THREAD

**Discussion by paulzakin** [procrastinate-org/procrastinate#735] [Q&A] [answered]
For example, you have a trigger that fires when a new row is inserted into table Foo. In the PL/PG trigger, could you execute a SQL statement like INSERT payload into procrastinate_table? The event (bar) would have already been defined in the Python code so this is really just a way to add a task via a trigger.

**Comment by ewjoachim** [comment, marked as answer]
Hello,

It should be possible using the procedures that procrastinate creates, though it's not part of the official API and as such it's undocumented. Note that it's unlikely to break suddenly due to our migration rules, but you'll need to test that it works as expected when you do a major upgrade (if there even is one)

https://github.com/procrastinate-org/procrastinate/blob/b7cfd09ed9fae7cc039501db9d0dda937845a3d5/procrastinate/sql/schema.sql#L72

**Reply by paulzakin** [reply]
Awesome, thank you!
